### PR TITLE
feat: add option to restrict the default resolver to IPv4 or IPv6

### PR DESCRIPTION
**Pull request by Garionion**
This PR adds the ability to restrict the default Resolver to IPv4 or IPv6 only. Its useful when the targets are only reachable by e.g. IPv4 anyway and one wants to reduce the load on the DNS Server

### CODE REVIEW

File: dnscache.go [hunk 1]
```diff
@@ -246,17 +246,45 @@ func (r *Resolver) storeLocked(key string, rrs []string, used bool, err error) {
...
+type defaultResolverWithTrace struct {
+	ipVersion string
+}
 
 func (d *defaultResolverWithTrace) LookupHost(ctx context.Context, host string) (addrs []string, err error) {
 	// `net.Resolver#LookupHost` does not cause invocation of `net.Resolver#lookupIPAddr`, therefore the `DNSStart` and `DNSDone` tracing hooks
 	// built into the stdlib are never called. `LookupIP`, despite it's name, can also be used to lookup a hostname but does cause these hooks to be
 	// triggered. The format of the reponse is different, therefore it needs this thin wrapper converting it.
-	rawIPs, err := net.DefaultResolver.LookupIP(ctx, "ip", host)
+	rawIPs, err := net.DefaultResolver.LookupIP(ctx, d.ipVersion, host)
```
Comment (rs) [inline dnscache.go:287]: You need to handle the case where d.ipVersion is "".

Comment (Garionion) [reply dnscache.go:287]: I would argue that this is not necessary as the only way to use this `defaultResolverWithTrace` (from outside this lib) is by either using the `defaultResolver` or by retreiving a `defaultResolverWithTrace` from the functions `NewResolverOnlyV4` and `NewResolverOnlyV6`. In all 3 cases there is alreade the ipVersion set.

Comment (rs) [reply dnscache.go:287]: Sure but code evolve over time and this situation could change.

Comment (Garionion) [reply dnscache.go]: ok i've added that if ipVersion is not ip, ip4 or ip6 then ip will be used

**Pull request merged**